nucliadb_node/src/shards/providers/unbounded_cache/unbounded_reader.rs

```diff
@@ -79,19 +85,20 @@ impl ShardReaderProvider for UnboundedShardReaderCache {
     fn load_all(&self) -> NodeResult<()> {
         let mut cache = self.write();
         let shards_path = self.shards_path.clone();
-        debug!("Recovering shards from {shards_path:?}...");
+
```

Comment:
:warning: Behaviour for `load_all` differed between the sync & async version:
- The async version loaded all shards and then grabbed the lock to insert all shards into cache. 
- The sync version held the lock while loading all shards.

This version behaves like the sync version and thus blocks the cache until everything is loaded. It seems safer but potentially a long block. The async version could be prone to race-conditions if a shard is opened while `load_all` is running. A third alternative would be to grab a write lock for each iteration of the loop, so other consumers get the chance to interact with the cache (albeit with high contention).